### PR TITLE
Adds base RoadNetwork/RoadGeometry Rust API.

### PR DESCRIPTION
**Pull request by francocipollone**
# 🎉 New feature

Related to #30 #31 

## Summary
Adds Rust API for a minor part of the maliput::api::RoadNetwork/RoadGeometry (cpp) interface.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

### CODE REVIEW

**Comment by francocipollone** [comment]
@agalbachicar Note that this case for the RoadNetwork and RoadGeometry is probably not the one that is related to the F2F talk about rust vs cpp ownership and heap allocation
In this case we have a road network being created in cpp world and we consume from there. Its allocaiton is in the head and managed by the cpp world.
It sounds that therefore makes sense ...:
 - Having a RoadNetwrok (Rust) that has a reference to a cpp binding:
    - `UniquePtr<maliput_sys::api::ffi::RoadNetwork>`
 - And having a RoadGeometry (rust) using a reference to the RoadGeometry cpp binding:
    - `&maliput_sys::api::ffi::RoadGeometry`


A different case happens with simpler structs/classes like LanePosition or InertialPosition. In those cases, having a cpp heap allocation sounds wrong and it is where I should narrow down the search for investigating that use case.

File: maliput/data/xodr/TShapeRoad.xodr
Comment (francocipollone) [inline maliput/data/xodr/TShapeRoad.xodr:1]: We could bring the xodr files from the maliput_malidrive binary installation. However I do think that from the Rust API perspective makes sense to have XODR files in the package. Otherwise the paths might be hard to follow: 
 - `maliput(rust) -> maliput-sdk -> maliput(bazel) -> maliput (repo where the xodr files' src code are located)`

Comment (agalbachicar) [reply maliput/data/xodr/TShapeRoad.xodr:1]: I'm OK to simplify now, but with a TODO.
I think we should not repeat resources and expose a variable to installed resources through the bindings.

File: maliput/examples/road_geometry.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    let maliput_malidrive_plugin_path = maliput_sdk::get_maliput_malidrive_plugin_path();
+    std::env::set_var("MALIPUT_PLUGIN_PATH", maliput_malidrive_plugin_path);
+
+    // Get location of odr resources
+    let package_location = std::env::var("CARGO_MANIFEST_DIR").unwrap();
+    let xodr_path = format!("{}/data/xodr/TShapeRoad.xodr", package_location);
+
+    let mut properties = HashMap::new();
+    properties.insert("road_geometry_id", "my_rg_from_rust");
+    properties.insert("opendrive_file", xodr_path.as_str());
```
Comment (agalbachicar) [inline maliput/examples/road_geometry.rs:44]:
```suggestion
    let road_network_properties = HashMap::from([("road_geometry_id", "my_rg_from_rust"), ("opendrive_file", xodr_path.as_str())]);
```

File: maliput/examples/road_geometry.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use maliput::api::RoadNetwork;
+use std::collections::HashMap;
```
Comment (agalbachicar) [inline maliput/examples/road_geometry.rs:32]: I think we can move these inside the main function, right?

**Review by agalbachicar — DISMISSED** [review]
LGTM

File: maliput/examples/road_geometry.rs (hunk 2 above)
Comment (francocipollone) [reply maliput/examples/road_geometry.rs]: Done

File: maliput/examples/road_geometry.rs (hunk 1 above)
Comment (francocipollone) [reply maliput/examples/road_geometry.rs]: Done

**Review by agalbachicar — APPROVED** [review]
LGTM

File: maliput/data/xodr/TShapeRoad.xodr
Comment (francocipollone) [reply maliput/data/xodr/TShapeRoad.xodr:1]: https://github.com/maliput/maliput-rs/issues/33
